crates/binstalk-fetchers/src/futures_resolver.rs

```diff
@@ -67,10 +69,18 @@ impl<T: Send + 'static, E: Send + 'static> FuturesResolver<T, E> {
     }
 
     /// Return the resolution.
-    pub fn resolve(self) -> impl Future<Output = Result<Option<T>, E>> {
+    pub fn resolve(self) -> impl Future<Output = Option<T>> {
         let mut rx = self.rx;
         drop(self.tx);
 
-        async move { rx.recv().await.transpose() }
+        async move {
+            while let Some(res) = rx.recv().await {
+                match res {
+                    Ok(ret) => return Some(ret),
+                    Err(err) => warn!(?err, "Fail to resolve the future"),
+                }
+            }
+            None
+        }
```

Comment:
This control flow feels a bit confusing, how about
```suggestion
        async move {
            loop {
	            match rx.recv().await {
	                Some(Ok(ret)) => return Some(ret),
	                Some(Err(err)) => warn!(?err, "Fail to resolve the future"),
	                None => return None,
	            }
            }
        }
```

Reply:
Thanks, that looks much better